woocommerce/woocommerce
# Revise default version selection in @wordpress/e2e-environment

### All Submissions:


### Changes proposed in this Pull Request:

One of the caveats to using `git` when finding the latest version of `wordpress`, `php`, and `mariadb` is that it is possible the `docker` image doesn't actually exist yet. This PR adds a new `bin/get-latest-docker-tag.js` file that uses the [Docker HTTP API](https://docs.docker.com/registry/spec/api/#listing-image-tags) to find the latest tag. I've also added support for restricting major versions to avoid sudden breakages in the environment.

### How to test the changes in this Pull Request:

1. After making sure the packages have been built using 
```
npm install
npm run build:packages
npm run docker:up
npm run test:e2e
npm run docker:down
```
2. Ensure that the versions for the downloaded images are the same or newer than the following:
- WordPress 5.5.1
- PHP 7.4.9
- MariaDB 10.5.5
3. Run the `tests/e2e/env/bin/get-latest-docker-tag.js` file for `php` and play with the name parameter. You can try out `7.2` and it will return the latest version of that release branch, etc.

### Other information:
